File: BookRental.Server/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace BookRental.Server.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }

        [Required(AllowEmptyStrings = false)]
        public required string Name { get; set; }

        [Required(AllowEmptyStrings = false)]
        public required string Synopsis { get; set; }

        [Required(AllowEmptyStrings = false)]
        public required string AuthorName { get; set; }

        public bool IsBorrowed { get; set; } = default;
    }
}

File: bookrental.client/src/app/model/book.ts
export class Book {
    id: number;
    name: string;
    synopsis: string;
    authorName: string;
    isBorrowed: boolean;

    constructor(private book : Book){
        this.id = book.id,
        this.name = book.name,
        this.synopsis = book.synopsis,
        this.authorName = book.authorName,
        this.isBorrowed = book.isBorrowed
    }
  }